src/Modules/LearningTasks/Tutor.LearningTasks.Tests/Integration/TaskProgressTests.cs

```diff
@@ -0,0 +1,248 @@
+        result.ShouldNotBeNull();
+        result.Id.ShouldBe(-3);
+        result.LearningTaskId.ShouldBe(-7);
+        result.StepProgresses.ShouldNotBeNull();
+        result.StepProgresses.Count.ShouldBe(1);
+        result.StepProgresses[0].ShouldNotBeNull();
+        result.StepProgresses[0].Id.ShouldBe(-3);
+        result.StepProgresses[0].Status.ShouldBe("Viewed");
+        result.StepProgresses[0].StepId.ShouldBe(-8);
+    }
+
+    [Fact]
+    public void Wrong_learner_views_step_returns_forbidden()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var actionResult = controller.ViewStep(-3, -5, -6).Result;
+        var objectResult = actionResult as ObjectResult;
+
+        dbContext.ChangeTracker.Clear();
+        objectResult.ShouldNotBeNull();
+        objectResult.StatusCode.ShouldBe(403);
+    }
+
+    [Fact]
+    public void Submits_answer()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var stepProgressDto = new StepProgressDto
+        {
+            Answer = "Answer",
+            StepId = -7
+        };
+
+        var actionResult = controller.SubmitAnswer(-2, -2, stepProgressDto).Result;
+        var okObjectResult = actionResult as OkObjectResult;
+        var result = okObjectResult?.Value as TaskProgressDto;
+
+        dbContext.ChangeTracker.Clear();
+        result.ShouldNotBeNull();
+        result.Id.ShouldBe(-2);
+        result.LearningTaskId.ShouldBe(-6);
+        result.Status.ShouldBe("Completed");
+        result.StepProgresses.ShouldNotBeNull();
+        result.StepProgresses.Count.ShouldBe(1);
+        result.StepProgresses[0].ShouldNotBeNull();
+        result.StepProgresses[0].Id.ShouldBe(-2);
+        result.StepProgresses[0].Answer.ShouldBe("Answer");
+        result.StepProgresses[0].Status.ShouldBe("Answered");
+        result.StepProgresses[0].StepId.ShouldBe(-7);
+    }
+
+    [Fact]
+    public void Wrong_learner_submits_answer_returns_forbidden()
+    {
+        using var scope = Factory.Services.CreateScope();
+        var controller = CreateController(scope);
+        var dbContext = scope.ServiceProvider.GetRequiredService<LearningTasksContext>();
+
+        var stepProgressDto = new StepProgressDto
+        {
+            Answer = "Answer",
+            StepId = -7
+        };
+
+        var actionResult = controller.SubmitAnswer(-3, -5, stepProgressDto).Result;
+        var objectResult = actionResult as ObjectResult;
+
+        dbContext.ChangeTracker.Clear();
+        objectResult.ShouldNotBeNull();
+        objectResult.StatusCode.ShouldBe(403);
+    }
+
+    [Fact]
+    public void Submits_answer_for_unexisting_learning_task()
```

Comment:
Using the insights from the comments on the previous test suite, rename roughly half of these tests.